File: app/src/main/java/mehdihaghgoo/khateratebook/CoverActivity.java
package mehdihaghgoo.khateratebook;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.AppCompatButton;
import android.view.View;
import android.widget.Button;

public class CoverActivity extends AppCompatActivity {

    Intent intent;
    public final static String EXTRA_MESSAGE="mehdihaghgoo.khateratebook.OPEN_MESSAGE";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_cover);

        Button btnShowTOC=(Button)findViewById(R.id.btn_show_toc_id);

    }

    public void showTOC(View view) {
        intent= new Intent(CoverActivity.this,MainActivity.class);
        intent.putExtra(EXTRA_MESSAGE,true);
        startActivity(intent);
    }
}

File: app/src/main/java/mehdihaghgoo/khateratebook/CoverActivity.kt
package mehdihaghgoo.khateratebook

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class CoverActivity : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_cover)
    }

    fun showTOC(view: View) {
        val intent: Intent ?= Intent(this, TOC_Activity::class.java)
        intent?.putExtra(EXTRA_MESSAGE, true)
        startActivity(intent)
    }

    companion object {
        val EXTRA_MESSAGE = "mehdihaghgoo.khateratebook.OPEN_MESSAGE"
    }
}
